synapseclient/api/table_services.py

```diff
@@ -123,6 +123,97 @@ async def get_columns(
     return columns
 
 
+async def get_column(
+    column_id: str,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Union[str, int]]:
+    """Get a single column by ID.
+
+    Arguments:
+        column_id: The ID of the column to retrieve.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The column matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/ColumnModel.html>
+    """
+    from synapseclient import Synapse
+
+    return await Synapse.get_client(synapse_client=synapse_client).rest_get_async(
+        f"/column/{column_id}",
+    )
+
+
+async def list_columns(
+    prefix: Optional[str] = None,
+    limit: int = 100,
+    offset: int = 0,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> AsyncGenerator["Column", None]:
+    """List columns with optional prefix filtering.
+
```

Comment:
It's not clear (to me) what this function does based on the docstring. Are these all the columns from a table?

Reply:
Not sure how you'd like me to update this. This function will list columns stored in Synapse. Synapse stores columns independently from tables. This is the same as how Activities in Synapse works. Tables then reference the IDs of columns to say "This table is using these columns"/